sycl/source/detail/graph_impl.cpp

```diff
@@ -382,11 +413,23 @@ graph_impl::add(const std::shared_ptr<graph_impl> &Impl,
   if (Handler.MSubgraphNode) {
     return Handler.MSubgraphNode;
   }
-  return this->add(Handler.MCGType, std::move(Handler.MGraphNodeCG), Dep);
+
+  node_type NodeType;
+  if (auto UserFacingType = Handler.MImpl->MUserFacingNodeType;
+      UserFacingType != node_type::empty) {
+    NodeType = UserFacingType;
+  } else {
+    NodeType = getNodeTypeFromCG(Handler.MCGType);
+  }
```

Comment:
Realized I commented in one place about the ternary, but not this one too. 
```
    ext::oneapi::experimental::node_type NodeType =
        MImpl->MUserFacingNodeType !=
                ext::oneapi::experimental::node_type::empty
            ? MImpl->MUserFacingNodeType
            : ext::oneapi::experimental::detail::getNodeTypeFromCG(MCGType);
```

Reply:
Fixed, good catch!